clients/cobol-lsp-vscode-extension/src/commands/SmartTabCommand.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import { DEFAULT_TAB_SIZE, PUNCH_CARD } from "../constants";
+import { ProfileService } from "../services/ProfileService";
+
+const SMART_TAB_COMMAND_NAME: string = "cobol-lsp.smart-tab";
+const BUILD_IN_TAB_COMMAND_NAME: string = "tab";
+
+let TabCommandHolder: vscode.Disposable = null;
+let ContextHolder: vscode.ExtensionContext = null;
+
+/**
+ * Initialize smart tab functionality
+ * @param context 
+ */
+export function initSmartTab(context: vscode.ExtensionContext) {
+    context.subscriptions.push(vscode.commands.registerTextEditorCommand(SMART_TAB_COMMAND_NAME, 
+    (editor: vscode.TextEditor, edit: vscode.TextEditorEdit, ...args: any[]) => {
+        executeSmartTab(editor, edit, args);
+    }));
+
+    ContextHolder = context;
+    performBinding();
+}
+
+/**
+ * Execute a smart tab command
+ */
+function executeSmartTab(editor: vscode.TextEditor, edit: vscode.TextEditorEdit, args: any[]) {
+    if (ProfileService.isCobolProgram(editor.document.uri.path)) {
```

Comment:
I think that you must not use it because this will lock tab stop usages only for COBOL extensions. What if I have a *.txt file and change the Language ID to COBOL?